src/jit/flowgraph.cpp

```diff
@@ -5158,6 +5218,8 @@ void Compiler::fgLinkBasicBlocks()
 
     /* Walk all the basic blocks, filling in the target addresses */
 
+    bool foundBackwardJump = false;
```

Comment:
`fgMarkBackwardJump` seems to be called when a recursive call is found in `impImportCall`. I believe recursive calls (except for explicit tail calls) aren't problematic because tiering would still be able to count the calls and tier-up in reasonable time. If it's possible for tier 0 JIT to generate a loop instead of recursion (when not an explicit tail call) then that could be problematic and may want to switch to tier 1 or avoid that optimization. When that would not happen I wanted to avoid switching to tier 1 for self-recursive methods. It looks like `impImportCall` would happen in a later stage so they wouldn't be caught here anyway. Is it because of potential implicit tail call that recursive calls are treated as loops?

Reply:
> Is it because of potential implicit tail call that recursive calls are treated as loops?

I'm not certain, but that would be my guess. Perhaps not now, but in future, we may want to refine when this is called (or what it does) for recursive calls based on whether it is likely to be transformed into a tail call. I'm nearly certain that we don't do any opportunistic (implicit) tail call transformations at tier0.